client/client.go

```diff
@@ -511,3 +549,44 @@ func (client *Client) DebugGet(action string, result interface{}, params map[str
 	_, err := client.doSync("GET", "/v1/debug", urlParams, nil, nil, &result)
 	return err
 }
+
+type defaultRequester struct {
+	baseURL   url.URL
```

Comment:
It looks like we're never setting the `baseURL` field -- oversight? We do use it in `dispatch` (somewhat surprised there's not a test for this -- we should probably add one).

Reply:
Hi Ben, correct me if I am missing something obvious but we have these:

```
if opts.BaseURL == "" {
		// By default talk over a unix socket.
		transport := &http.Transport{Dial: unixDialer(opts.Socket), DisableKeepAlives: opts.DisableKeepAlive}
		baseURL := url.URL{Scheme: "http", Host: "localhost"}
		requester = &defaultRequester{baseURL: baseURL, transport: transport}
:

```

See the last line of the snippet :)

Reply:
I think this is OK ? I created an issue to look at refactoring the client test framework to better utilise the Requester API.